### PR TITLE
Fix MTEBEvaluator: device mapping, padding-free inference, last-token pooling, L2 normalization

### PR DESCRIPTION
**Pull request by natke**
Fixes several issues in the MTEBEvaluator for embedding model evaluation:

## Device mapping
Maps Olive's `Device.GPU` (`"gpu"`) to PyTorch's `"cuda"` when initializing `SentenceTransformer` in the HF evaluation path.

## Padding-free inference for GenAI
GenAI's `Generator` does not accept an `attention_mask`, so padded batches produce contaminated hidden states via self-attention to padding tokens. Fix: process each sentence individually with only its real tokens, eliminating padding entirely.

## Last-token pooling
Replaced mean pooling with last-token pooling in the GenAI and ORT wrappers to match models like Qwen3-Embedding that use `pooling_mode_lasttoken=True`.

## L2 normalization
Added L2 normalization after pooling in the base `encode()` method, matching the `2_Normalize` module in the SentenceTransformer pipeline.

### Results
These fixes close the score gap between HF and GenAI evaluation:
- **Before**: HF 0.785 vs GenAI 0.651 (STS17 main_score)
- **After**: HF 0.785 vs GenAI 0.785

### CODE REVIEW

File: olive/evaluator/olive_evaluator.py [hunk 1]
```diff
@@ -1192,7 +1192,9 @@ def evaluate(
         if model_class == "hf":
             from sentence_transformers import SentenceTransformer
 
-            sentence_transformer_device = device.value if isinstance(device, Device) else str(device)
+            # Map Olive Device to PyTorch device string (Olive uses "gpu", PyTorch expects "cuda")
+            device_str = device.value if isinstance(device, Device) else str(device)
+            sentence_transformer_device = "cuda" if device_str == "gpu" else device_str
```
[the remark this answers is not in the document]
Comment (natke) [reply olive/evaluator/olive_evaluator.py]: fixed

**Comment by natke** [comment]
Closing fork PR — recreating from microsoft/Olive branch directly (now a contributor).
